bidscoin/bids.py

```diff
@@ -1310,7 +1310,7 @@ def find_run(bidsmap: dict, provenance: str, dataformat: str='', datatype: str='
                     return run
 
 
-def delete_run(bidsmap: dict, provenance: Union[dict, str], datatype: str= '') -> None:
+def delete_run(bidsmap: dict, provenance: Union[dict, str], datatype: str= '', dataformat: str='') -> None:
```

Comment:
This is all it took to for me to be able to clean up items that were collected by multiple plugins. Tried to stay away from library code, but couldn't come up with a better way to do this action.

Reply:
That is fine with me, but my first thought was: The user could also avoid this by setting PET as `exclude` datatype in the dcm2niix2bids plugin, and have it converted by your plugin?

Reply:
I mean exclude it in the DICOM tab (which is handled by the dcm2niix2bids plugin), but not in the PET tab

Reply:
Alternatively, if you change the order of the plugins, the dcm2niix4petbids will get executed first, and the dcm2niix2bids plugin will just skip the PET run-item (as it then already exists in the bidsmap)?

Reply:
Excluding is not deleting (in one tab), but that can be a bit confusing for the user indeed. Same holds for the order of the plugins (that can handle the same source data type), but another way out is to remove PET handling altogether from dcm2niix2bids? I mean, it was just a quick and simple hack, better to replace it altogether with a separate plugin?

Reply:
I am happy to limit dcm2niix2bids to MRI only. Have a separate plugin for MRS and another one for PET

Reply:
> Excluding is not deleting (in one tab), but that can be a bit confusing for the user indeed. Same holds for the order of the plugins (that can handle the same source data type), but another way out is to remove PET handling altogether from dcm2niix2bids? I mean, it was just a quick and simple hack, better to replace it altogether with a separate plugin?

You've sold me. I just wanted to make sure that we're not losing any functionality, which in theory we shouldn't.

Reply:
I now skip processing DICOM PET data in the dcm2niixplugin (see commit 8a577db2966e564f6a24f252106d35eb9eea8437), So I suppose you can now just delete the PET section in your template bidsmap and use the PET entries in the DICOM section. To do so, in the dcm2niix4pet plugin you should change your dataformat `PET` (which is not a dataformat anyway) to `DICOM`

Reply:
> I now skip processing DICOM PET data in the dcm2niixplugin (see commit [8a577db](https://github.com/Donders-Institute/bidscoin/commit/8a577db2966e564f6a24f252106d35eb9eea8437)), So I suppose you can therefore delete the PET section in your template bidsmap and use the PET entries in the DICOM section. To do so, in the dcm2niix4pet plugin you should change your data format `PET` (which is not a data format anyway) to `DICOM`

I was using PET as I figured that `PET` would be better able to capture ECAT, Dicoms w/ modality 'PT', and spreadsheets that contain PET metadata. I haven't integrated handling spreadsheets or ECAT with this PR as I wanted to keep it as small as possible while I got familiar with the codebase. So the question is would you rather have a `PET` data format or add `ECAT` and `Spreadsheet`?  I'm biased towards the former obviously.